src/ui/class-post-states.php

```diff
@@ -41,11 +52,24 @@ public function register_hooks() {
 	 * @return array The updated post states array.
 	 */
 	public function show_original_in_post_states( $post_states, $post ) {
-		$original_item = Utils::get_original( $post );
-		if ( $original_item ) {
-			/* translators: Original item link (to view or edit) or title. */
+		$original_item                 = Utils::get_original( $post );
+		$is_rewrite_and_republish_copy = $this->permissions_helper->is_rewrite_and_republish_copy( $post );
+
+		if ( ! $original_item ) {
+			return $post_states;
+		}
+
+		if ( $is_rewrite_and_republish_copy ) {
```

Comment:
I would move `$this->permissions_helper->is_rewrite_and_republish_copy( $post )` here without introducing a `$is_rewrite_and_republish_copy` variable, so the check is also done *only* if we are dealing with a copy. Otherwise it will be done for every element in the post list, even for those which are not copies at all...